parse.c

```diff
@@ -71,10 +71,11 @@ static void show_option_range(const struct fio_option *o,
 			      size_t (*logger)(const char *format, ...))
 {
 	if (o->type == FIO_OPT_FLOAT_LIST) {
-		if (o->minfp == DBL_MIN && o->maxfp == DBL_MAX)
+		if (!o->minfp && !o->maxfp)
 			return;
 
-		logger("%20s: min=%f", "range", o->minfp);
+		if (o->minfp != DBL_MIN)
+			logger("%20s: min=%f", "range", o->minfp);
```

Comment:
If you were to only set o->maxfp the logger output might look strange

Reply:
Can you clarify your comment? Either both minfp and maxfp should be set or none of them so I don't see the issue.

Reply:
Imagine I want a range from 0 - DBL_MAX then won't I only see ", max=..." in the logs?

Reply:
With the following patch:

diff --git a/options.c b/options.c
index 17d7245f5490..3cd84bfe0608 100644
--- a/options.c
+++ b/options.c
@@ -3504,7 +3511,7 @@ struct fio_option fio_options[FIO_MAX_OPTS] = {
 		.def	= "100",
 		.maxlen	= 1,
 		.minfp	= 0.0,
-		.maxfp	= 100.0,
+		.maxfp	= DBL_MAX,
 		.category = FIO_OPT_C_IO,
 		.group	= FIO_OPT_G_LATPROF,
 	},

fio shows the following output:

./fio --cmdhelp=latency_percentile
  latency_percentile: Percentile of IOs must be below latency_target
                type: list of floating point values separated by ':' (opt=5.9:7.8)
             default: 100
               range: min=0.000000

I think that's the output we all expect?

Reply:
I had it the wrong way round - o->minfp would have to be DBL_MIN and o->maxfp would have to be something other than DBL_MAX:

patch:
```patch
diff --git a/options.c b/options.c
index 17d7245f..0b7c3076 100644
--- a/options.c
+++ b/options.c
@@ -6,6 +6,7 @@
 #include <assert.h>
 #include <sys/stat.h>
 #include <netinet/in.h>
+#include <float.h>
 
 #include "fio.h"
 #include "verify.h"
@@ -3503,7 +3504,7 @@ struct fio_option fio_options[FIO_MAX_OPTS] = {
                .help   = "Percentile of IOs must be below latency_target",
                .def    = "100",
                .maxlen = 1,
-               .minfp  = 0.0,
+               .minfp  = DBL_MIN,
                .maxfp  = 100.0,
                .category = FIO_OPT_C_IO,
                .group  = FIO_OPT_G_LATPROF,
```
example:
```
$ ./fio --cmdhelp=latency_percentile
  latency_percentile: Percentile of IOs must be below latency_target
                type: list of floating point values separated by ':' (opt=5.9:7.8)
             default: 100
, max=100.000000
```

Reply:
See also https://github.com/axboe/fio/pull/574

Reply:
Yup that got it.